From 40985902afd8f4eb0bd7aa4238c7d458282af948 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Wed, 21 Dec 2016 08:20:18 +0100
Subject: [PATCH] ivshmem-net: Improve identification of resources

Pass a device name consisting of driver name and PCI ID to request_irq
and alloc_ordered_workqueue. This helps correlating resources with
devices in case there are multiple of them.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
(cherry picked from commit 43e2ff78b89cbdfaecba54601d85f3d40349a9b5)
---
 drivers/net/ivshmem-net.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

--- a/drivers/net/ivshmem-net.c
+++ b/drivers/net/ivshmem-net.c
@@ -762,6 +762,7 @@ static int ivshm_net_probe(struct pci_de
 	resource_size_t shmaddr;
 	resource_size_t shmlen;
 	int interrupt;
+	char *device_name;
 	void *shm;
 	u32 ivpos;
 	int err;
@@ -814,7 +815,10 @@ static int ivshm_net_probe(struct pci_de
 		return -EINVAL;
 	}
 
-	dev_info(&pdev->dev, "shared memory size %pa\n", &shmlen);
+	device_name = devm_kasprintf(&pdev->dev, GFP_KERNEL, "%s[%s]", DRV_NAME,
+				     dev_name(&pdev->dev));
+	if (!device_name)
+		return -ENOMEM;
 
 	ndev = alloc_etherdev(sizeof(*in));
 	if (!ndev)
@@ -837,7 +841,7 @@ static int ivshm_net_probe(struct pci_de
 	if (err)
 		goto err_free;
 
-	in->state_wq = alloc_ordered_workqueue(DRV_NAME, 0);
+	in->state_wq = alloc_ordered_workqueue(device_name, 0);
 	if (!in->state_wq)
 		goto err_free;
 
@@ -866,7 +870,7 @@ static int ivshm_net_probe(struct pci_de
 		in->using_msix = false;
 	}
 
-	err = request_irq(interrupt, ivshm_net_int, 0, DRV_NAME, ndev);
+	err = request_irq(interrupt, ivshm_net_int, 0, device_name, ndev);
 	if (err)
 		goto err_int;
 
